x-pack/test/functional/apps/uptime/locations.ts

```diff
@@ -28,21 +30,26 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
         }
         return d;
       };
-      await makeChecksWithStatus(
-        getService('legacyEs'),
-        MONITOR_ID,
-        5,
-        2,
-        10000,
-        {},
-        'up',
-        mogrifyNoLocation
-      );
+      const esService = getService('legacyEs');
+      await makeChecksWithStatus(esService, MONITOR_ID, 5, 2, 10000, {}, 'up', mogrifyNoLocation);
+      await makeChecksWithStatus(esService, 'filter-testing', 50, 1, 10000, {}, 'up', (d: any) => {
+        d.observer.geo.name = 'filter-test';
+        return d;
+      });
     });
 
     it('renders the location missing popover when monitor has location name, but no geo data', async () => {
       await pageObjects.uptime.loadDataAndGoToMonitorPage(start, end, MONITOR_ID);
       await pageObjects.uptime.locationMissingIsDisplayed();
     });
+
+    it('adds a second location filter option and filters the overview', async () => {
+      await pageObjects.uptime.goToUptimePageAndSetDateRange(start, end);
+      await pageObjects.uptime.selectFilterItems({ location: ['filter-test'] });
+      await retry.tryForTime(12000, async () => {
+        const snapshotCount = await pageObjects.uptime.getSnapshotCount();
+        expect(snapshotCount).to.eql({ up: '1', down: '0' });
```

Comment:
Is there any sort of test we could do on the ping histogram to catch breakage there? Looking around I don't see anything super obvious to hook into.

Reply:
Addressed in https://github.com/elastic/kibana/pull/56794/commits/6952a02dbc37a859a9b9ebabb46661724d05a7b4, https://github.com/elastic/kibana/pull/56794/commits/98923d4631b3762623983749f0eefc458bf97755, https://github.com/elastic/kibana/pull/56794/commits/08288732fcfd0825220a4e3de3f13f965b3a2787, and https://github.com/elastic/kibana/pull/56794/commits/3b3462a8f8b327cd0f110ccd3813514d3d7ea936.